consensus/src/pipeline/execution_client.rs

```diff
@@ -215,10 +217,9 @@ impl ExecutionProxyClient {
                 let (rand_ready_block_tx, rand_ready_block_rx) = unbounded::<OrderedBlocks>();
 
                 let (reset_tx_to_rand_manager, reset_rand_manager_rx) = unbounded::<ResetRequest>();
-                let consensus_key =
-                    load_consensus_key_from_secure_storage(&self.consensus_config.safety_rules)
-                        .expect("Failed in loading consensus key for ExecutionProxyClient.");
-                let signer = Arc::new(ValidatorSigner::new(self.author, consensus_key));
+                let consensus_sk = maybe_consensus_key
+                    .expect("consensus key unavailable for ExecutionProxyClient");
```

Comment:
given we just panic if the option doesn't exist, we should probably pass in Arc directly?

Reply:
we unwrap only if the current node is in the validator set.
For non-validators i guess we can't unwrap outside...